### DISCUSSION TITLE
TableauServerClient downgraded in 2.9?

### DISCUSSION THREAD

**Discussion by scottbw** [apache/airflow#41880] [General]
Hi,

In Airflow 2.6.3, the constraints file (https://raw.githubusercontent.com/apache/airflow/constraints-2.6.3/constraints-3.10.txt) has:

tableauserverclient==0.24

While in Airflow 2.9.2, the constraints file (https://raw.githubusercontent.com/apache/airflow/constraints-2.9.2/constraints-3.11.txt) has:

tableauserverclient==0.19.0

Why the downgrade? Also, the current version of TSC, that works with current server API versions, is 0.32.

**Comment by potiuk** [comment]
Likely conflicting with other dependencies in this particular combinations. Analysing every single of 719 dependencies of ours for reasoning would be a herculean effort but if you want you can attempt to install alrflow in this version and upgrade the tableauserver and see what it conflicts with (it could be some extras so it might be not an easy task) 

But my suggestion is to upgrade airflow to 2.9.3 or better - 2.10.1 (will be released shortly) and there seems the conflict is gone.

**Comment by potiuk** [comment]
BTW. you are also free to upgrade tsc to this version in your image/installation - constraints are just "hints" on how to install airlfow + all providers in the way that will not conflict with all other dependencies (in reproducible way), but you are absolutely free (and it is recommended) to individually upgrade or downgrade dependencies as you need. It's described in all our installation docs for PyPI installation including various upgrade scenarios.
